File: backend/app/schemas/notification.py
from datetime import datetime
from pydantic import BaseModel
from app.schemas.common import BaseSchema


class CreatorRef(BaseSchema):
    id: str
    name: str


class NotificationCreate(BaseModel):
    type: str = "announcement"
    title: str
    message: str
    audience_role: str | None = None
    expires_at: datetime | None = None


class NotificationUpdate(BaseModel):
    title: str | None = None
    message: str | None = None
    is_published: bool | None = None


class NotificationOut(BaseSchema):
    id: str
    type: str
    title: str
    message: str
    audience_role: str | None = None
    is_published: bool
    published_at: datetime | None = None
    expires_at: datetime | None = None
    created_by_user_id: str | None = None
    created_by_user: CreatorRef | None = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/app/core/models/notification.ts
export type NotificationType = 'announcement' | 'holiday' | 'policy_update' | 'leave_update' | 'system';

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  audience_role: string | null;
  is_published: boolean;
  published_at: string | null;
  expires_at: string | null;
  created_at: string;
  created_by_user?: { name: string } | null;
}
